src/neo/SmartContract/ApplicationEngine.Contract.cs

```diff
@@ -161,10 +162,9 @@ private void CallContractInternal(ContractState contract, ContractMethodDescript
             CallFlags callingFlags = state.CallFlags;
 
             if (args.Count != method.Parameters.Length) throw new InvalidOperationException($"Method {method.Name} Expects {method.Parameters.Length} Arguments But Receives {args.Count} Arguments");
-            ExecutionContext context_new = LoadScript(contract.Script, method.Offset);
+            ExecutionContext context_new = LoadScript(contract.Script, flags & callingFlags, contract.ScriptHash, method.Offset);
```

Comment:
Is it really needed to merge flags `flags & callingFlags`?

Reply:
We need to remove the flags that are not present in the parent